### PR TITLE
SAML: Add name id configs

### PR DESCRIPTION
**Pull request by tung2744**
ref DEV-1729
requires https://github.com/authgear/authgear-server/pull/4573

### CODE REVIEW

**Comment by louischan-oursky** [comment]
Please rebase.

**Review by louischan-oursky — APPROVED** [review]

File: pkg/lib/config/saml.go [hunk 1]
```diff
@@ -19,7 +19,9 @@ var _ = Schema.Add("SAMLServiceProviderConfig", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"id": { "type": "string" }
+		"id": { "type": "string" },
+		"name_id_format": { "$ref": "#/$defs/SAMLNameIDFormat" },
```
Comment (louischan-oursky) [inline pkg/lib/config/saml.go:23]: NameID should be `nameid`

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

File: pkg/lib/config/saml.go (hunk 1 above)
Comment (tung2744) [reply pkg/lib/config/saml.go:23]: The PascelCase NameID should be name_id in snake_case?

**Comment by tung2744** [comment]
Sorry I forgot to update the metadata, marking as WIP

File: pkg/lib/config/saml.go (hunk 1 above)
Comment (louischan-oursky) [reply pkg/lib/config/saml.go:23]: Since it is `rn:oasis:names:tc:SAML:1.1:nameid-format:unspecified`, I think `nameid` is one word.

Comment (tung2744) [reply pkg/lib/config/saml.go:23]: I see. Ok let me change it.

**Comment by tung2744** [comment]
Updated, thanks!

**Comment by tung2744** [comment]
I've marked here https://github.com/authgear/authgear-server/pull/4513#issuecomment-2288013004

File: pkg/lib/saml/error.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package saml
+
+import "errors"
+
+var ErrServiceProviderNotFound = errors.New("service provider not found")
```
[the remark this answers is not in the document]
Comment (tung2744) [reply pkg/lib/saml/error.go:5]: Actually in saml endpoints, we need to send the error to the acs url with a `<Response>`. So it is still incorrect to return a api error which returns a json in the current request.
I plan to handle this part in https://linear.app/authgear/issue/DEV-1727/%5Bsamlbrowsersso%5D-return-a-response-with-redirect-binding

[the remark this answers is not in the document]
Comment (louischan-oursky) [reply pkg/lib/saml/error.go:5]: Even with this metadata endpoint? Why not return a `<Response>` directly in this metadata endpoint?

[the remark this answers is not in the document]
Comment (tung2744) [reply pkg/lib/saml/error.go:5]: In metadata endpoint, ErrServiceProviderNotFound was catched and treated as not found error (404).
Usually, metadata endpoint should not have any error. The only error here is 404. Returning any other thing is incorrect.

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**